sklearn/model_selection/_validation.py

```diff
@@ -298,7 +359,62 @@ def cross_validate(
     else:
         scorers = _check_multimetric_scoring(estimator, scoring)
 
-    indices = cv.split(X, y, groups)
+    if _routing_enabled():
+        # `cross_validate` will create a `_MultiMetricScorer` if `scoring` is a
+        # dict at a later stage. We need the same object for the purpose of
+        # routing. However, creating it here and passing it around would create
+        # a much larger diff since the dict is used in many places.
```

Comment:
This mentions that if we create _MultiMetricScorer scorer here it would create a large diff. Is this intended?

Reply:
yes, the two issues I opened are related to this diff. This is for future developers, not users, so it's nice to have it here as a reminder as why the code is as is.